src/UserCodeRunner.ts

```diff
@@ -390,9 +390,13 @@ export class UserCodeRuntimeError extends UserCodeError {
 
 	public get location(): { line: number; column: number } {
 		const stack = parse(this.error);
+		const userFileStackFrame = stack.find(callSite => callSite.getFileName() === USER_CODE_FILENAME);
+		if (userFileStackFrame === undefined) {
+			throw new Error('Runtime error detected outside of user code execution path. This is most likely a bug in the additional library source.');
+		}
 		const originalPosition = this.sourceMap.originalPositionFor({
-			line: stack[0].getLineNumber()!,
-			column: stack[0].getColumnNumber()!,
+			line: userFileStackFrame.getLineNumber()!,
```

Comment:
I forget - do we just return the location, or the whole stack trace?

Reply:
We return the stack trace as it pertains to user code (so removing any harness and additional file stack information) as well.